### PR TITLE
Mw/net 5679 explicit upstreams mesh webhook injects upstream env variable

### PR DESCRIPTION
**Pull request by wilkermichael**
Changes proposed in this PR:
- This introduces injection of the port/ip address environment variables

How I've tested this PR:
- unit tests

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by DanStough — CHANGES_REQUESTED** [review]
Looks great! My only dealbreaker was the naming of the ENV variable. Open to discussion, though.

File: control-plane/connect-inject/common/annotation_processor.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	return &upstream, nil
+}
+
+func UpstreamReferenceType() *pbresource.Type {
+	return &pbresource.Type{
+		Group:        "catalog",
+		GroupVersion: "v1alpha1",
+		Kind:         "Service",
+	}
+}
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply control-plane/connect-inject/common/annotation_processor.go]: oh yeah, I'll fix this after Iryna merges her changes in

File: control-plane/connect-inject/constants/constants_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+package constants
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetDefaultConsulNamespace(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply control-plane/connect-inject/constants/constants_test.go]: ah thanks, good catch

File: control-plane/connect-inject/webhook_v2/container_env.go [hunk 3]
```diff
@@ -4,34 +4,38 @@
...
+
+	var result []corev1.EnvVar
+	for _, upstream := range upstreams.Upstreams {
+		serviceName := strings.TrimSpace(upstream.DestinationRef.Name)
+		serviceName = strings.ToUpper(strings.Replace(serviceName, "-", "_", -1))
+		portName := strings.TrimSpace(upstream.DestinationPort)
+		portName = strings.ToUpper(strings.Replace(portName, "-", "_", -1))
+
+		result = append(result, corev1.EnvVar{
+			Name:  fmt.Sprintf("%s_%s_CONNECT_SERVICE_HOST", serviceName, portName),
```
[the remark this answers is not in the document]
Comment (DanStough) [reply control-plane/connect-inject/webhook_v2/container_env.go:32]: My bad, the RFC is just like how you have it :homer-disappear:

File: control-plane/connect-inject/webhook_v2/container_env.go [hunk 4]
```diff
@@ -4,34 +4,38 @@
...
-	//			Name:  fmt.Sprintf("%s_CONNECT_SERVICE_HOST", name),
-	//			Value: "127.0.0.1",
-	//		}, corev1.EnvVar{
-	//			Name:  fmt.Sprintf("%s_CONNECT_SERVICE_PORT", name),
-	//			Value: portStr,
-	//		})
-	//	}
-	//}
+func (w *MeshWebhook) containerEnvVars(pod corev1.Pod) ([]corev1.EnvVar, error) {
+	upstreams, err := common.ProcessPodUpstreams(pod, true, true)
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply control-plane/connect-inject/webhook_v2/container_env.go:16]: it doesn't have enablePartitions, and also honestly it doesn't really matter for this usecase. We don't care about the partition/namespace values (they're optional anyways if they're enabled) so we can just enable it always.

**Comment by wilkermichael** [comment]
@DanStough can you take a look at this now that I've addressed your comments? Thanks!

**Review by thisisnotashwin — APPROVED** [review]

**Review by DanStough — APPROVED** [review]

**Pull request merged**